packages/world/src/systems/LogoffSystem.sol

```diff
@@ -15,16 +15,15 @@ import { ActionType } from "../codegen/common.sol";
 import { MIN_TIME_BEFORE_AUTO_LOGOFF } from "../Constants.sol";
 import { AirObjectID, PlayerObjectID } from "../ObjectTypeIds.sol";
 import { getUniqueEntity } from "../Utils.sol";
-import { requireValidPlayer } from "../utils/PlayerUtils.sol";
+import { requireValidPlayer, deletePlayer } from "../utils/PlayerUtils.sol";
 import { notify, LogoffNotifData } from "../utils/NotifUtils.sol";
 
 import { EntityId } from "../EntityId.sol";
 
 contract LogoffSystem is System {
   function logoffCommon(EntityId playerEntityId, VoxelCoord memory playerCoord) internal {
     LastKnownPosition._set(playerEntityId, playerCoord.x, playerCoord.y, playerCoord.z);
-    PlayerPosition._deleteRecord(playerEntityId);
-    ReversePlayerPosition._deleteRecord(playerCoord.x, playerCoord.y, playerCoord.z);
+    deletePlayer(playerEntityId, playerCoord);
```

Comment:
ideally deletePlayer shouldn't receive a playerCoord, as we can get everything from the entity.

Wdyt of not returning the coord from `requireValidPlayer`? It is not really used in that function

Reply:
Hmm, but `playerCoord` is used in almost all places right after the `requireValidPlayer` call, even in this logoff call need it for `LastKnownPosition`

Reply:
right, but it feels to me (subjective) that it has a bit of semantic overload. requireValidPlayer is already doing a bunch of stuff other than "requiring it is a valid player". And due to returning the coord, we end up with having to pass the coord to `deletePlayer`, when `deletePlayer` could just receive the entityId and get the coord internally

Reply:
hmm playerCoord could be moved out, but then in all the places `requireValidPlayer` is called, right after we'll need to get the coord. Even in logoff, need the player coord before calling `deletePlayer` to set `LastKnownPosition`

also I like having one spot to run the things that should happen on each tx: drain player energy, set active timestamp

Maybe we should rename `requireValidPlayer` because yea its doing more than "requiring it is a valid player". How about like `requireValidPlayer` only does the checks, and then another util `getPlayerInfo` calls `requireValidPlayer` and in addition does the other 3 things: 1) player coord, 2) update energy, 3) set `PlayerActivity`

Reply:
I prefer just keeping `requireValidPlayer` over that for now,  if we have something like `getPlayerInfo` I would expect it not to modify the state

Reply:
I understand the idea of grouping stuff that is usually done together, but sometimes that obscures too much what the code is doing, but we can leave for now and revisit later :)